src/AP.AzureADAuth/ViewModels/LoginPageViewModel.cs

```diff
@@ -41,18 +41,19 @@ public LoginPageViewModel(IAuthenticationService authenticationService, IEventAg
         public async void OnAppearing()
         {
             _logger.TrackEvent("Login Page");
+            var canExecuteCommand = await LoginCommand.CanExecute.FirstAsync();
 
-            await LoginCommand.Execute();
+            if (canExecuteCommand)
+            {
+                await LoginCommand.Execute();
+            }
         }
-
         public void OnDisappearing() { }
 
         public void Destroy()
         {
             _isBusyHelper.Dispose();
             _isBusyHelper = null;
-            LoginCommand.Dispose();
-            LoginCommand = null;
```

Comment:
why is this being removed?

Reply:
Actually, i was facing an exception"System.ObjectDisposedException: 'Cannot access a disposed object.'" once the LoginCommand gets disposed. It seems framework is trying to access it after its disposal. That's why it commented out these lines and removed them later.